src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
-
-    if input_args.indexed:
-        ids['toil.bam.bai'] = job.addChildJobFn(download_url, "%s.bai" % url, 'toil.bam.bai').rv()
-        job.addFollowOnJobFn(haplotype_caller, ids, input_args, cores = input_args.cpu_count)
+        for uuid, gvcf in gvcfs.iteritems():
+            job.addFollowOnJobFn(hard_filter_pipeline,
+                                 uuid, gvcf,
+                                 config)
+
+
+def gatk_germline_pipeline(job, uuid, url, config, rg_line=None):
+    """
+    Runs GATK Germline Pipeline on a single sample. Writes gvcf to output directory
+    defined in the config dictionary. Removes secondary alignments.
+
+    0: Align FASTQ or Download Bam
+    1: Generate BAI
+    2: GATK Preprocessing (Optional)
+    3: GATK HaplotypeCaller
+    4: Write GVCF to output directory
+
+    :param JobFunctionWrappingJob job: Toil Job instance
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to sample bam or fq file
+    :param Namespace config: Configuration options for pipeline
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: GVCF FileStoreID
+    :rtype: str
+    """
+    config = deepcopy(config)
+    config.uuid = uuid
+    config.url = url
+    config.rg_line = rg_line
+
+    cores = multiprocessing.cpu_count()
+
+    # Determine how much RAM is available
+    if config.xmx is None:
+        config.xmx = os.sysconf('SC_PAGE_SIZE') * os.sysconf('SC_PHYS_PAGES')
+
+    get_bam = job.wrapJobFn(prepare_bam, uuid, url, config, rg_line=rg_line).encapsulate()
+
+    job.addChild(get_bam)
+    input_bam = get_bam.rv(0)
+    input_bai = get_bam.rv(1)
+
+    if config.preprocess:
+        preprocess = job.wrapJobFn(run_germline_preprocessing,
+                                   cores,
+                                   get_bam.rv(0),
+                                   get_bam.rv(1),
+                                   config.genome_fasta,
+                                   config.genome_dict,
+                                   config.genome_fai,
+                                   config.phase,
+                                   config.mills,
+                                   config.dbsnp,
+                                   xmx=config.xmx).encapsulate()
+        get_bam.addChild(preprocess)
+        if config.preprocess_only:
+            output_dir = os.path.join(config.output_dir, uuid)
+            filename = '{}.{}.bam'.format(uuid, config.suffix)
+            output_bam = job.wrapJobFn(upload_or_move_job,
+                                       filename,
+                                       preprocess.rv(0),
+                                       output_dir)
+            preprocess.addChild(output_bam)
+            return preprocess.rv(0), preprocess.rv(1)
+        else:
+            input_bam = preprocess.rv(0)
+            input_bai = preprocess.rv(1)
+
+    hc_disk = PromisedRequirement(lambda x: 2*x.size + human2bytes('5G'), input_bam)
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.